contracts/EVMScriptFactories/TopUpAllowedRecipients.sol

```diff
@@ -0,0 +1,159 @@
+// SPDX-FileCopyrightText: 2022 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.4;
+
+import "../TrustedCaller.sol";
+import "../AllowedRecipientsRegistry.sol";
+import "../interfaces/IFinance.sol";
+import "../libraries/EVMScriptCreator.sol";
+import "../interfaces/IEVMScriptFactory.sol";
+import "../EasyTrack.sol";
+
+/// @notice Creates EVMScript to top up allowed recipients addresses within the current spendable balance
+contract TopUpAllowedRecipients is TrustedCaller, IEVMScriptFactory {
+    // -------------
+    // ERRORS
+    // -------------
+    string private constant ERROR_LENGTH_MISMATCH = "LENGTH_MISMATCH";
+    string private constant ERROR_EMPTY_DATA = "EMPTY_DATA";
+    string private constant ERROR_ZERO_AMOUNT = "ZERO_AMOUNT";
+    string private constant ERROR_RECIPIENT_NOT_ALLOWED = "RECIPIENT_NOT_ALLOWED";
+    string private constant ERROR_SUM_EXCEEDS_SPENDABLE_BALANCE = "SUM_EXCEEDS_SPENDABLE_BALANCE";
+
+    // -------------
+    // VARIABLES
+    // -------------
+
+    /// @notice Address of EasyTrack contract
+    EasyTrack public immutable easyTrack;
+
+    /// @notice Address of Aragon's Finance contract
+    IFinance public immutable finance;
+
+    /// @notice Address of payout token
+    address public immutable token;
```

Comment:
Let's not go with `immutable` here? Would be extremely painful to verify different deployments otherwise.